src/excalidraw-app/collab/Collab.tsx

```diff
@@ -551,12 +562,12 @@ class Collab extends PureComponent<Props, CollabState> {
       }
     | { fetchScene: false; roomLinkData?: null }) => {
     clearTimeout(this.socketInitializationTimer!);
-    if (this.portal.socket && this.fallbackInitializationHandler) {
-      this.portal.socket.off(
-        "connect_error",
-        this.fallbackInitializationHandler,
-      );
+
+    if (!this.portal.socket || this.portal.socketInitialized) {
+      return null;
     }
```

Comment:
we don't want to init if no socket exists or socket already initialized.

In any case, we'll need to rewrite Collab & Portal so that we create new Portal instance for each room (and destroy it on room close), and we don't retain any room state past Portal close (and we shouldn't be passing room data to these functions as that has the tendency of operating on a potentially already closed rooms, as we were in this case).